File: IRPlayer-swift/Class/Core/Display/OpenGL/Program/Factory/IRGLProgram2DFactory.m
//
//  IRGLProgram2DFactory.m
//  IRPlayer
//
//  Created by Phil on 2019/7/5.
//  Copyright © 2019 Phil. All rights reserved.
//

#import "IRGLProgram2DFactory.h"
#import <IRPlayer_swift/IRPlayer_swift-Swift.h>

@implementation IRGLProgram2DFactory

-(IRGLProgram2D*)createIRGLProgramWithPixelFormat:(IRPixelFormat)pixelFormat withViewprotRange:(CGRect)viewprotRange withParameter:(nullable IRMediaParameter*)parameter{
    return [IRGLProgramFactory createIRGLProgram2DWithPixelFormat:pixelFormat viewportRange:viewprotRange parameter:(IRMediaParameter*)parameter];
}

@end

File: Sources/IRPlayer-swift/Class/Core/Display/OpenGL/Program/Factory/IRGLProgram2DFactory.swift
//
//  IRGLProgram2DFactory.swift
//  IRPlayer-swift
//
//  Created by irons on 2024/9/13.
//

import Foundation

public class IRGLProgram2DFactory: NSObject {

    func createIRGLProgram(pixelFormat: IRPixelFormat, viewportRange: CGRect, parameter: IRMediaParameter?) -> IRGLProgram2D {
        return IRGLProgramFactory.createIRGLProgram2D(pixelFormat: pixelFormat, viewportRange: viewportRange, parameter: parameter)
    }
}

